src/Dapr.Workflow/WorkflowServiceCollectionExtensions.cs

```diff
@@ -41,12 +39,35 @@ public static class WorkflowServiceCollectionExtensions
             }
 
             serviceCollection.TryAddSingleton<WorkflowRuntimeOptions>();
-            serviceCollection.TryAddSingleton<WorkflowClient>();
+            serviceCollection.TryAddSingleton<WorkflowEngineClient>();
             serviceCollection.AddDaprClient();
 
+            serviceCollection.AddOptions<WorkflowRuntimeOptions>().Configure(configure);
+
+            static bool TryGetGrpcAddress(out string address)
```

Comment:
We have a shared package that may already do what you need here.

https://github.com/dapr/dotnet-sdk/blob/master/src/Shared/DaprDefaults.cs

Reply:
Thanks for pointing this out, I hadn't seen it. I ran into a couple problems trying to use it as-is though. For now, I'm going to put a TODO comment in the code suggesting that we adopt the common code if/when the problems can be safely addressed.